ex_0811/Person.java

```diff
@@ -0,0 +1,93 @@
+package ex_0811;
+
+import java.util.Scanner;
+
+public class Person {
+    private int id;
+    private String name;
+    private String gender;
+    private String dateBirth;
+    private String email;
+
+    public Person() {
+
+    }
+
+    public Person(int id, String name, String gender, String dateBirth, String email) {
+        this.id = id;
+        this.name = name;
+        this.gender = gender;
+        this.dateBirth = dateBirth;
+        this.email = email;
+    }
+
+    public int getId() {
+        return id;
+    }
+
+    public boolean setId(int id) {
+        if (id > 0) {
+            this.id = id;
+            return true;
+        } else {
+            System.out.println("Please enter id again: ");
+            return false;
+        }
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public boolean setName(String name) {
+        if (name != null) {
+            this.name = name;
+            return true;
+        } else {
+            System.out.println("Please enter name again: ");
+            return false;
+        }
+    }
+
+    public String getGender() {
+        return gender;
+    }
+
+    public boolean setGender(String gender) {
+        if (gender.equalsIgnoreCase("male") || gender.equalsIgnoreCase("female")) {
+            this.gender = gender;
+            return true;
+        } else {
+            System.out.println("Please enter gender again: ");
+            return false;
+        }
+    }
+
+    public String getDateBirth() {
+        return dateBirth;
+    }
+
+    public boolean setDateBirth(String dateBirth) {
+        if (dateBirth != null) {
+            this.dateBirth = dateBirth;
+            return true;
+        } else {
+            System.out.println("Please enter date of birth again: ");
+            return false;
+        }
+    }
+
+    public String getEmail() {
+        return email;
+    }
+
+    public boolean setEmail(String email) {
+        if (email != null && email.contains("@") && !email.contains(" ")) {
```

Comment:
good, can look for Regular Expression to learn about string pattern. But this idea is ok